tests/test_text.py

```diff
@@ -0,0 +1,464 @@
+# -*- coding: utf-8 -*-
+# Copyright FMR LLC <opensource@fidelity.com>
+# SPDX-License-Identifier: GNU GPLv3
+import random
+import selectors
+import pandas as pd
+from feature.selector import Selective, SelectionMethod
+from test_base import BaseTest
+from textwiser import TextWiser, Embedding, Transformation
+
+
+class TestText(BaseTest):
```

Comment:
Lots of good tests here! My only high-level comment is
1) not following some of the if-else none pass code
2) we are not verfiying the selection. It would be good to capture the exact selected items so that we can reproduce the tests identically. Notice the size of the data is small 4 columns or so, hence returning a data with 2 subsets is very likely evevn if the 2 selected is "incorrect" 

For Random, we can assert to select the same indices with the same seed. You can actually push a "best" solution, and even its random, it should end up selecting  the best index 

For Greedy, you can constrcut labels such that you know the greedy columsn. Verfiy that they are selected at the end. 

For Kmeans, it is not obvious to me how to do that index verifications. I would just run it once, and then assert the same selection from this moment onward (with the same seed). 

For Exact, same idea. run once and then assert the same selection of indices going forward. 

BTW, are we sure to test most (if not all?) combinations? I know there are too many parameterization with/without features (2) X num_opt (4) X num_cost 2 = 16. EVEN IF we are not asserting anything, just running each combination at least ONCE is a test in and of itself.

Reply:
Regarding the if-else none pass code, I made changes to simplify the code to make it more readable and easier to understand. 

I agree that it would be good to capture the exact selected items so that we can reproduce the tests identically.
For Random, I have added an assert statement to verify that the same indices are selected with the same seed. I also added a comment suggesting that for the "random" optimization method, we can push for a "best" solution, and even though it is random, it should end up selecting the best indices. 

For Greedy, I have constructed labels to know the greedy columns and verified that they are selected at the end. We should set the seed for Greedy as there is a source of randomness in the initialization of the Lagrangian multiplier.

For KMeans and Exact, we have run the tests once and verified that the same selection of indices is chosen going forward with the same seed. For Kmeans, we need to set the seed as follows: 
seed = seed value
np.random.seed(seed)
random.seed(seed) 
sklearn.set_config(working_memory=0) sklearn.utils.check_random_state(seed)
The reason is that Kmeans uses random initialization of cluster centroids. Even with the same random seed, the starting position of centroids can differ due to various factors such as the optimization algorithm used, data normalization, etc. I set the random seed of NumPy and scikit-learn packages to ensure reproducibility. This will ensure that the results are reproducible across multiple runs.

I have tested all possible combinations of features, optimization methods, and cost metrics to ensure that all combinations are tested at least once.

Reply:
"We should set the seed for Greedy as there is a source of randomness in the initialization of the Lagrangian multiplier." 

Is this captured in the PyDoc of TextBased in selector.py

Reply:
Added to the PyDoc for greedy